### PR TITLE
Me/serializer

### PR DESCRIPTION
**Pull request by meysholdt**

### CODE REVIEW

**Comment by meysholdt** [comment]
Jenkins xtext-runtime-tests-master-latest

File: plugins/org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/XtextInjectorProvider.java [hunk 1]
```diff
@@ -8,22 +8,22 @@
 
 import com.google.inject.Injector;
 
-public class XtextInjectorProvider implements IInjectorProvider, IRegistryConfigurator
-{
+/**
+ * @since 2.9
+ */
+public class XtextInjectorProvider implements IInjectorProvider, IRegistryConfigurator {
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/XtextInjectorProvider.java]: This injector provider is used to test Xtext itself. I don't think it should be part of the juni4 test library.

File: plugins/org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/ContextProvider.java [hunk 2]
```diff
@@ -32,64 +22,22 @@
  * @author Moritz Eysholdt - Initial contribution and API
  */
 @Singleton
+@Deprecated
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/ContextProvider.java]: What's supposed to be used as an alternative?

**Comment by meysholdt** [comment]
Jenkins test

File: plugins/org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/XtextInjectorProvider.java (hunk 1 above)
Comment (meysholdt) [inline plugins/org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/XtextInjectorProvider.java]: I moved it to the o.e.xtext.junit4.internal package

File: plugins/org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/ContextProvider.java (hunk 2 above)
Comment (meysholdt) [inline plugins/org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/ContextProvider.java]: there is JavaDoc now

**Comment by meysholdt** [comment]
Jenkins xtext-runtime-tests-master-latest

**Comment by meysholdt** [comment]
Jenkins xtext-runtime-tests-master-latest

**Pull request merged**